### ISSUE TITLE
Large negative numbers on the Y axis are not abbreviated

### ISSUE THREAD

**Issue by kizmiaz7** [imaNNeoFighT/fl_chart#486]
Hello, 

Great work on all the updates man, and thank you for implementing the Y axis interval feature I requested last year! 👍 👍  

I have a line graph that has large positive and negative Y values, and I noticed the large negative values aren't shortened (1,000,000 => 1M for example). I am using an interval, for the left Side Tiles. 

I have attached a screenshot for you reference. I'm assuming this is unintended. Would it be possible to 'fix'?

Thanks!

![Screen Shot 2020-10-29 at 2 16 57 PM](https://user-images.githubusercontent.com/16199436/97545780-7a9f3900-19f1-11eb-93b7-0f164592fa69.png)

**Comment by LondonTran** [comment]
I'm wondering the same thing. After looking through the Fl_Chart package files, I noticed axis_chart_data.dart line 288: 
`getTitles = getTitles ?? defaultGetTitle,`

When I CTRL then click on _defaultGetTitle_ it takes me to

base_chart_data.dart line 132 
```
String defaultGetTitle(double value) {
  return formatNumber(value);
}
```
and then CTRL then clicking on _formatNumber_ finally takes me to 
utils.dart line 155
```
String formatNumber(double number) {
  String resultNumber;
  String symbol;
  if (number >= billion) {
    resultNumber = (number / billion).toStringAsFixed(1);
    symbol = 'B';
  } else if (number >= million) {
    resultNumber = (number / million).toStringAsFixed(1);
    symbol = 'M';
  } else if (number >= kilo) {
    resultNumber = (number / kilo).toStringAsFixed(1);
    symbol = 'K';
  } else {
    resultNumber = number.toStringAsFixed(1);
    symbol = '';
  }

  if (resultNumber.endsWith('.0')) {
    resultNumber = resultNumber.substring(0, resultNumber.length - 2);
  }

  return resultNumber + symbol;
}
```

Looking back at `getTitles = getTitles ?? defaultGetTitle,` it appears that the number formatter isn't being activated because _getTitles_ is not null so _defaultGetTitle_ isn't reached. Does anyone know how we can use the _formatNumber(double number)_ method? 

I tried reversing it to `getTitles = defaultGetTitle ?? getTitles,` but this does not work and prints an axis title for every value.

**Comment by imaNNeoFighT** [comment]
I feel good to fixed your last year's problem :)
about the bug, you are right, we didn't support large negative numbers. we must edit the `formatNumber()` function.
I will fix this ASAP.
Thanks!

**Comment by imaNNeoFighT** [comment]
@LondonTran if you have getTitles function implemented, it means you are responsible for representing numbers. (you can format them if you want, BTY it's your responsibility).
And if you want, you can just copy/paste the implemented `formatNumber()` function to your code.
Thanks!

**Comment by imaNNeoFighT** [comment]
Fixed in [0.12.1](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.12.1). Check it out!.
Thanks for reporting!
